boards/arm/stm32l496g_disco/doc/stm32l496g_disco.rst

```diff
@@ -0,0 +1,255 @@
+- Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell™
+
+
+More information about STM32L496AG can be found here:
+       - `STM32L496AG on www.st.com`_
+       - `STM32L496 reference manual`_
+
+Supported Features
+==================
+
+The Zephyr stm32l496g_disco board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/stm32l496g_disco/stm32l496g_disco_defconfig``
+
+
+Connections and IOs
+===================
+
+STM32L496G Discovery Board has 8 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+For mode details please refer to `STM32L496G Discovery board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_1_TX : PB6
+- UART_1_RX : PG10
+- UART_2_TX : PA2
+- UART_2_RX : PD6
+- I2C_1_SCL : PB8
+- I2C_1_SDA : PB7
+- PWM_2_CH1 : PA0
+- LD2 : PB13
+
+System Clock
+------------
+
+STM32L496G Discovery System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by PLL clock at 80MHz,
+driven by 16MHz high speed internal oscillator.
+
+Serial Port
+-----------
+
+STM32L496G Discovery board has 5 U(S)ARTs. The Zephyr console output is assigned to UART2.
+Default settings are 115200 8N1.
+
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+
+STM32L496G Discovery board includes an ST-LINK/V2-1 embedded debug tool interface.
+This interface is not supported by the openocd version 0.9 included by the Zephyr SDK v0.9,
+use openocd v0.10.0 from the openocd-stm32 project on GitHub to get the minimum set of scripts
+needed to flash and debug STM32 development boards.
+
+.. code-block:: console
+
+   $ git clone  https://github.com/erwango/openocd-stm32.git
```

Comment:
Indeed, I created this repo since openocd version proposed by Zephyr SDK did not support STM32L4 SoCs.
It just adds few add-on scripts on top of openocd for ease of use.
Last time I checked, there was no add-on script for F7 though (but supported by openocd core)